src/context_free.ml

```diff
@@ -478,6 +478,15 @@ class map_top_down ?(expect_mismatch_handler = Expect_mismatch_handler.nop)
       map_node EC.pattern pattern super#pattern x.ppat_loc base_ctxt x
 
     method! expression base_ctxt e =
+      let base_ctxt, e =
+        (* Make sure code-path attribute is applied before expanding. *)
+        match Attribute.get Ast_traverse.enter_value e with
+        | None -> (base_ctxt, e)
+        | Some { loc; txt } ->
+            ( Expansion_context.Base.enter_value ~loc txt base_ctxt,
+              Attribute.remove_seen Expression [ T Ast_traverse.enter_value ] e
```

Comment:
Have you seen the new `Attribute.remove_seen_res`? @panglesd has done the very nice job of making sure the ppxlib driver embeds errors instead of raising. What do you think about doing that here as well?